### ISSUE TITLE
shrunk values from Arb.numericDoubles do not respect the given range

### ISSUE THREAD

**Issue by sielenk-yara** [kotest/kotest#2377]
using Kotest 4.6.1

A similar problem as mentioned in #2284 exists for `Arb.numericDoubles`.

I am using `Arb.numericDoubles(from = 1e-6, to = 1e6)` to generate non-zero factors and on failure the shrinker tries 0.0:

    Attempting to shrink arg 504245.91197859275
    Shrink #1: 0.0 fail
    Shrink result (after 1 shrinks) => 0.0

**Comment by sielenk-yara** [comment]
A test to trigger the problem:
```
   test("shrunk Arb.numericDouble values should stay within bounds") {
      val min = 12363.0
      val max = 772183.0
      val generator = Arb.numericDouble(min = min, max = max)
      val (v, s) = generator.sample(RandomSource.seeded(39084345))
      val collector = mutableListOf(v)

      doShrinking(s, ShrinkingMode.Bounded(100)) {
         collector.add(it)
         it shouldBe 0.0
      }

      collector shouldHaveLowerBound min
      collector shouldHaveUpperBound max
   }
```

**Comment by sksamuel** [comment]
Fixed in 5.0
